starter-code/scripts/models/article.js

```diff
@@ -91,6 +97,16 @@
       the matching articles written by the specified author. */
     return Article.allAuthors().map(function(author) {
       return {
+        name: author,
+        numWords: Article.allArticles.filter(function(current){
+          return current.author === author;
+        })
+        .map(function(article){
+          return article.body.split(' ').length;
+        })
+        .reduce(function(current, previous){
+          return current + previous;
+        })
```

Comment:
good job!. when using reduce, always keep in mind what parameters go in, and in which order. i hope the morning reduce() lecture helped!